### PR TITLE
Zone compatibility changes

### PR DESCRIPTION
**Pull request by YegorFeoktistov**
I made a mistake in description of first commit, I meant ZoneShape class instead of Point class.

### CODE REVIEW

File: src/zone/shrinkSteps.ts [hunk 1]
```diff
@@ -2,9 +2,9 @@
...
  * @description Interface for object, that contains steps counters
  */
-export interface ShrinkSteps {
-  topStep: number,
-  bottomStep: number,
-  leftStep: number,
-  rightStep: number
-}
\ No newline at end of file
+export interface IShrinkSteps {
```
Comment (MarozauArtsiom) [inline src/zone/shrinkSteps.ts]: why interface was renamed?

File: src/zone/zone.ts [hunk 2]
```diff
@@ -37,16 +29,18 @@ export class Zone {
...
    * @this {Zone}
-   * @description Constructor of the Zone class */
+   * @description Constructor of the Zone class
+   */
   public constructor(shrinkCoefficient: number, lastZoneSide: number) {
     this._isFirstStage = true;
     this._isNewStage = true;
-    this._finalZoneShape = new ZoneShape();
-    this._currentZoneShape = new ZoneShape();
+    this._finalZoneShape = new ZoneShape(new Point(0, 0), new Point(0, 0));
```
Comment (MarozauArtsiom) [inline src/zone/zone.ts]: use default parameters values

File: src/zone/zone.ts [hunk 3]
```diff
@@ -251,10 +224,25 @@ export class Zone {
    * @description Calculates distances between zones
    */
   private calculateDistances(): void {
-    this._topDistance = Math.abs(this._finalZoneShape.upperLeftPoint.y - this._currentZoneShape.upperLeftPoint.y);
-    this._bottomDistance = Math.abs(this._currentZoneShape.lowerRightPoint.y - this._finalZoneShape.lowerRightPoint.y);
-    this._leftDistance = Math.abs(this._finalZoneShape.upperLeftPoint.x - this._currentZoneShape.upperLeftPoint.x);
-    this._rightDistance = Math.abs(this._currentZoneShape.lowerRightPoint.x - this._finalZoneShape.lowerRightPoint.x);
+    this._topDistance = Math.abs(
```
Comment (MarozauArtsiom) [inline src/zone/zone.ts]: make calculation on single line

File: src/zone/zone.ts [hunk 4]
```diff
@@ -251,10 +224,25 @@ export class Zone {
...
+    );
+
+    this._bottomDistance = Math.abs(
+      this._currentZoneShape.lowerRightPoint.y -
+      this._finalZoneShape.lowerRightPoint.y
+    );
+
+    this._leftDistance = Math.abs(
+      this._finalZoneShape.upperLeftPoint.x -
+      this._currentZoneShape.upperLeftPoint.x
```
Comment (MarozauArtsiom) [inline src/zone/zone.ts]: the same for this and others in this file

File: src/zone/zone.ts [hunk 5]
```diff
@@ -363,53 +351,53 @@ export class Zone {
...
-  private shrinkHorizontally(location: Array<any>, fillingObject: any, shrinkSteps: ShrinkSteps): void {
-    const isLeftSideReached = (this._currentZoneShape.upperLeftPoint.x === this._finalZoneShape.upperLeftPoint.x);
-    const isRightSideReached = (this._currentZoneShape.lowerRightPoint.x === this._finalZoneShape.lowerRightPoint.x);
+  private shrinkHorizontally(shrinkSteps: IShrinkSteps): void {
+    const isLeftSideReached = (
+      this._currentZoneShape.upperLeftPoint.x ===
+      this._finalZoneShape.upperLeftPoint.x
+    );
+    const isRightSideReached = (
+      this._currentZoneShape.lowerRightPoint.x ===
```
Comment (MarozauArtsiom) [inline src/zone/zone.ts]: single line

File: src/zone/zone.ts [hunk 6]
```diff
@@ -424,72 +412,75 @@ export class Zone {
...
           shrinkSteps.rightStep++;
           this._horizontalStepCount = 0;
         }
-      }
-      else if (this._leftDistance < this._rightDistance) {
+      } else if (this._leftDistance < this._rightDistance) {
         shrinkSteps.rightStep++;
 
-        ZoneFilling.unequalDistancesHorizontalLoop(location, fillingObject, upperY, lowerY, lowerX, upperX, isCommonStep);
+        // ZoneFilling.unequalDistancesHorizontalLoop(battlefield, fillingObject, upperY, lowerY, lowerX, upperX, isCommonStep);
```
Comment (MarozauArtsiom) [inline src/zone/zone.ts]: are you sure that this code comments are necessary?

File: src/zone/zone.ts [hunk 7]
```diff
@@ -104,9 +94,8 @@ export class Zone {
   }
   public set shrinkCoefficient(value: number) {
     if (value <= 0) {
-      throw new Error("Invalid shrinking coefficient. The value should be greater then 0.");
-    }
-    else {
+      throw new Error('Invalid shrinking coefficient. The value should be greater then 0.');
```
Comment (MarozauArtsiom) [inline src/zone/zone.ts]: did it mean, that if bot send incorrect data then game cycle will break?

Comment (YegorFeoktistov) [reply src/zone/zone.ts]: No. When you create zone instanse with wrong parameters, it will throw an error.

Comment (YegorFeoktistov) [reply src/zone/zone.ts]: Bot does not interact with the zone algorithm

**Pull request merged**